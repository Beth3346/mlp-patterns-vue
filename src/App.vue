<template>
  <div id="app">
    <header class="header-main">
      <h1><router-link to="/">MyLittlePonyCollection Pattern Library</router-link></h1>
    </header>
    <div id="nav">
      <router-link to="/elements">Elements</router-link> |
      <router-link to="/compounds">Compounds</router-link>
    </div>
    <main class="main-content">
      <router-view/>
    </main>
  </div>
</template>

<style lang="scss">
@import '/sass/_config.scss';

#app {
  max-width: 1200px;
  margin: auto;
}

#nav {
  padding: 10px 0;
}

.main-content {
  margin: 20px 0;
}

.header-main {
  padding: 10px 0;

  h1 {
    color: #ef0c71;
    font-size: 24px;
    font-weight: 900;

    a {
      text-decoration: none;
    }
  }
}
</style>
