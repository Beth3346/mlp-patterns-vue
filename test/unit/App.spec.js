import { shallow } from '@vue/test-utils';
import helpers from 'elr-vue-jest-helpers';
import App from '@/App.vue';

let wrapper;
let h;

beforeEach(() => {
  wrapper = shallow(App);
  h = helpers.init({
    wrapper,
    expect
  });
});

describe('App.vue', () => {
  it('renders correctly', () => {
    h.matches('.header-main');
  });
  it('renders a nav', () => {
    h.matches('#nav');
  });
});
