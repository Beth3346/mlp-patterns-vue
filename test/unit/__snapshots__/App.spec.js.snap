// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue renders a nav 1`] = `
<div id="nav">
    <router-link to="/elements">Elements</router-link> |
    <router-link to="/compounds">Compounds</router-link>
</div>
`;

exports[`App.vue renders correctly 1`] = `
<header class="header-main">
    <h1>
        <router-link to="/">MyLittlePonyCollection Pattern Library</router-link>
    </h1>
</header>
`;
